src/AppInstallerRepositoryCore/Microsoft/CheckpointIndex.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "SQLiteWrapper.h"
+#include "Microsoft/Schema/ICheckpointIndex.h"
+#include "Microsoft/SQLiteStorageBase.h"
+#include <winget/ManagedFile.h>
+
+namespace AppInstaller::Repository::Microsoft
+{
+    struct CheckpointIndex : SQLiteStorageBase
+    {
+        // An id that refers to a specific Checkpoint.
+        using IdType = SQLite::rowid_t;
+
+        CheckpointIndex(const CheckpointIndex&) = delete;
+        CheckpointIndex& operator=(const CheckpointIndex&) = delete;
+
+        CheckpointIndex(CheckpointIndex&&) = default;
+        CheckpointIndex& operator=(CheckpointIndex&&) = default;
+
+        // Creates a new CheckpointIndex database of the given version.
+        static CheckpointIndex CreateNew(const std::string& filePath, Schema::Version version = Schema::Version::Latest());
+
+        // Opens an existing CheckpointIndex database.
+        static CheckpointIndex Open(const std::string& filePath, OpenDisposition disposition, Utility::ManagedFile&& indexFile = {})
+        {
+            return { filePath, disposition, std::move(indexFile) };
+        }
+
+        // Opens or creates a CheckpointIndex database on the default path.
+        static std::shared_ptr<CheckpointIndex> OpenOrCreateDefault(GUID guid, OpenDisposition openDisposition = OpenDisposition::ReadWrite);
```

Comment:
I don't really see any scenario where the disposition would be anything but `ReadWrite`.  I also don't see any scenario where you would want create *or* open; every new resumable should be create and every resume command should be open.